cookbooks/aws-parallelcluster-install/recipes/intel_mpi.rb

```diff
@@ -17,40 +17,46 @@
 
 return unless node['conditions']['intel_mpi_supported']
 
-intelmpi_modulefile = "/opt/intel/impi/#{node['cluster']['intelmpi']['version']}/intel64/modulefiles/intelmpi"
-intelmpi_installer_archive = "l_mpi_#{node['cluster']['intelmpi']['version']}.tgz"
-intelmpi_installer_path = "#{node['cluster']['sources_dir']}/#{intelmpi_installer_archive}"
-intelmpi_installer_url = "https://#{node['cluster']['region']}-aws-parallelcluster.s3.#{node['cluster']['region']}.#{aws_domain}/archives/impi/#{intelmpi_installer_archive}"
+intelmpi_installation_path = "/opt/intel/mpi/#{node['cluster']['intelmpi']['version']}"
+intelmpi_modulefile = "#{intelmpi_installation_path}/modulefiles/intelmpi"
+intelmpi_installer = "l_mpi_oneapi_p_#{node['cluster']['intelmpi']['full_version']}_offline.sh"
+intelmpi_installer_path = "#{node['cluster']['sources_dir']}/#{intelmpi_installer}"
+intelmpi_installer_url = "https://#{node['cluster']['region']}-aws-parallelcluster.s3.#{node['cluster']['region']}.#{aws_domain}/archives/impi/#{intelmpi_installer}"
 
 # fetch intelmpi installer script
 remote_file intelmpi_installer_path do
   source intelmpi_installer_url
   mode '0744'
   retries 3
   retry_delay 5
-  not_if { ::File.exist?("/opt/intel/impi/#{node['cluster']['intelmpi']['version']}") }
+  not_if { ::File.exist?(intelmpi_installation_path.to_s) }
 end
 
 bash "install intel mpi" do
   cwd node['cluster']['sources_dir']
   code <<-INTELMPI
     set -e
-    tar -xf #{intelmpi_installer_archive}
-    cd l_mpi_#{node['cluster']['intelmpi']['version']}/
-    ./install.sh -s silent.cfg --accept_eula
-    mv rpm/EULA.txt /opt/intel/impi/#{node['cluster']['intelmpi']['version']}
-    cd ..
-    rm -rf l_mpi_#{node['cluster']['intelmpi']['version']}*
+    chmod +x #{intelmpi_installer}
+    ./#{intelmpi_installer} --remove-extracted-files yes -a --silent --eula accept --install-dir /opt/intel
+    rm -f #{intelmpi_installer}
   INTELMPI
-  creates "/opt/intel/impi/#{node['cluster']['intelmpi']['version']}"
+  creates intelmpi_installation_path.to_s
 end
 
 append_if_no_line "append intel modules file dir to modules conf" do
   path node['cluster']['modulepath_config_file']
-  line "/opt/intel/impi/#{node['cluster']['intelmpi']['version']}/intel64/modulefiles/"
+  line "#{intelmpi_installation_path}/modulefiles/"
```

Comment:
Further improvement (we could also do it later on): determine the directory from the variable `intelmpi_modulefile `

Reply:
I'm skipping this for now to avoid re-run of build image